From 2b832a5471b3b94a0406ecf86119388ab6bbf122 Mon Sep 17 00:00:00 2001
From: Greg Hudson <ghudson@mit.edu>
Date: Wed, 6 Jun 2012 19:13:13 -0400
Subject: Don't free caller's principal in vfy_increds

but left in the corresponding free, so it was freeing a caller-owned
principal.  Reported by Russ Allbery.

ticket: 7162
(cherry picked from commit dd64191e02df0a13b29345e4c50fe03e039dc207)

Patch-Category: upstream
---
 src/lib/krb5/krb/vfy_increds.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/src/lib/krb5/krb/vfy_increds.c b/src/lib/krb5/krb/vfy_increds.c
index bd993c2..32ebbb9 100644
--- a/src/lib/krb5/krb/vfy_increds.c
+++ b/src/lib/krb5/krb/vfy_increds.c
@@ -316,7 +316,6 @@ cleanup:
 
     if (defkeytab != NULL)
         krb5_kt_close(context, defkeytab);
-    krb5_free_principal(context, server);
     free_princ_list(context, host_princs);
 
     return ret;
